lib/models/LifecycleConfiguration.ts

```diff
@@ -501,15 +501,27 @@ export default class LifecycleConfiguration {
     _checkDate(date: string) {
         const isoRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
             '(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9])' +
-            ':([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z)?$');
+            ':([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z|[+-][0-5][0-9]:[0-5][0-9])?$');
         if (!isoRegex.test(date)) {
             const msg = 'Date must be in ISO 8601 format';
             return errors.InvalidArgument.customizeDescription(msg);
         }
-        const midnightRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
-            '(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T00' +
-            ':00:00(\\.0+)?(Z)?$');
-        if (!midnightRegex.test(date)) {
+        // Extract the time portion of the date string to check if a timezone
+        // is specified. If not, add a Z to indicate UTC. We split to avoid
+        // then hyphens in the date portion.
+        const time = date.split('T')[1];
+        if (!time.includes('Z') && !time.includes('+') && !time.includes('-')) {
+            date += 'Z';
```

Comment:
Would it make more sense to use match groups in the previous regexp, so we can reuse them here?

Reply:
Sounds much better, I'll look into that

Reply:
Done, thanks!